lib/internal/crypto/random.js

```diff
@@ -454,79 +462,38 @@ function generatePrime(size, options, callback) {
 
     callback(
       undefined,
-      bigint ?
-        BigInt(`0x${Buffer.from(prime).toString('hex')}`) :
-        prime);
+      job.result(prime));
   };
   job.run();
 }
 
 function generatePrimeSync(size, options = {}) {
   validateUint32(size, 'size', true);
-  validateObject(options, 'options');
-  const {
-    safe = false,
-    bigint = false,
-  } = options;
-  let {
-    add,
-    rem,
-  } = options;
-  validateBoolean(safe, 'options.safe');
-  validateBoolean(bigint, 'options.bigint');
-
-  if (add !== undefined) {
-    if (typeof add === 'bigint') {
-      add = Buffer.from(toHexPadded(add), 'hex');
-    } else if (!isAnyArrayBuffer(add) && !isArrayBufferView(add)) {
-      throw new ERR_INVALID_ARG_TYPE(
-        'options.add',
-        [
-          'ArrayBuffer',
-          'TypedArray',
-          'Buffer',
-          'DataView',
-          'bigint',
-        ],
-        add);
-    }
-  }
-
-  if (rem !== undefined) {
-    if (typeof rem === 'bigint') {
-      rem = Buffer.from(toHexPadded(rem), 'hex');
-    } else if (!isAnyArrayBuffer(rem) && !isArrayBufferView(rem)) {
-      throw new ERR_INVALID_ARG_TYPE(
-        'options.rem',
-        [
-          'ArrayBuffer',
-          'TypedArray',
-          'Buffer',
-          'DataView',
-          'bigint',
-        ],
-        rem);
-    }
-  }
 
-  const job = new RandomPrimeJob(kCryptoJobSync, size, safe, add, rem);
+  const job = createRandomPrimeJob(kCryptoJobSync, size, options);
   const [err, prime] = job.run();
   if (err)
     throw err;
+  return job.result(prime);
+}
 
-  return bigint ?
-    BigInt(`0x${Buffer.from(prime).toString('hex')}`) :
-    prime;
+function arrayBufferToUnsignedBigInt(arrayBuffer) {
+  return BigInt(`0x${Buffer.from(arrayBuffer).toString('hex')}`);
 }
 
-function toHexPadded(bigint) {
+function unsignedBigIntToBuffer(bigint, name) {
+  if (bigint < 0) {
+    throw new ERR_OUT_OF_RANGE(name, '>= 0', bigint);
+  }
+
   const hex = bigint.toString(16);
```

Comment:
Explanation: The value `hex` begins with a minus sign if `bigint < 0`, causing `Buffer.from(hex, 'hex')` to return an empty buffer, that OpenSSL treats as a `0`.